microfocus_smax/operations.py

```diff
@@ -0,0 +1,212 @@
+""" Copyright start
+  Copyright (C) 2024 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+
+import requests, json
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.core.utils import update_connnector_config
+
+logger = get_logger('microfocus_smax')
+
+
+class MicroFocusSmax(object):
+    def __init__(self, config):
+        self.server_url = config.get("server_url", "").strip("/")
+        self.verify_ssl = config.get('verify_ssl')
+        self.username = config.get("username")
+        self.password = config.get("password")
+        self.tenant_id = config.get("tenant_id")
+        self.token = self.get_token()
+
+        self.connector_info = config.get("connector_info")
+        if self.connector_info:
+            update_connnector_config(connector_name=self.connector_info.get("connector_name"),
+                                     version=self.connector_info.get("connector_version"),
+                                     updated_config=config,
+                                     configId=config.get("config_id"))
+
+    def get_token(self):
+        try:
+            headers = {
+                'Content-Type': 'application/json'
+            }
+            data = {
+                "login": self.username,
+                "password": self.password
+            }
+            params = {
+                "TENANTID": self.tenant_id
+            }
+            login_api = "/auth/authentication-endpoint/authenticate/login"
+            url = self.server_url + login_api
+            response = requests.request(method="POST", url=url,
+                                        headers=headers, json=data, params=params, verify=self.verify_ssl)
+            if response.ok:
+                return response.text
+            else:
+                raise ConnectorError(json.loads(response.text).get("message"))
+        except Exception as e:
+            if 'Max retries exceeded' in str(e):
+                raise ConnectorError(
+                    'Max retries exceeded. Please check the URL provided for configuration of connector')
+            raise ConnectorError(e)
+
+    def make_api_call(self, method="GET", params=None, data=None, headers=None, url=None, json_data=None):
+        try:
+            url = self.server_url + url
+            if json_data is None:
+                json_data = {}
+
+            if headers:
+                headers.update({"Cookie" : "SMAX_AUTH_TOKEN="+self.token})
+            else:
+                headers = {"Cookie": "SMAX_AUTH_TOKEN=" + self.token}
```

Comment:
make curl is missing
  try:
      from connectors.debug_utils.curl_script import make_curl
      make_curl(method, request_url, headers=headers, params=payload, data=data, verify_ssl=self.verify_ssl)
  except Exception:
      pass